plugin/trino-hive/src/main/java/io/trino/plugin/hive/aws/athena/projection/DateProjectionFactory.java

```diff
@@ -0,0 +1,201 @@
+                columnProperties,
+                COLUMN_PROJECTION_FORMAT,
+                String::valueOf);
+
+        List<String> range = getProjectionPropertyRequiredValue(
+                columnName,
+                columnProperties,
+                COLUMN_PROJECTION_RANGE,
+                value -> ((List<?>) value).stream()
+                        .map(String.class::cast)
+                        .collect(toImmutableList()));
+        if (range.size() != 2) {
+            throw invalidRangeProperty(columnName, dateFormatPattern);
+        }
+
+        SimpleDateFormat dateFormat = new SimpleDateFormat(dateFormatPattern);
+        dateFormat.setLenient(false);
+        dateFormat.setTimeZone(TimeZone.getTimeZone(UTC_TIME_ZONE_ID));
+        Supplier<Instant> leftBound = parseDateRangerBound(columnName, range.get(0), dateFormat);
+        Supplier<Instant> rangeBound = parseDateRangerBound(columnName, range.get(1), dateFormat);
+        if (!leftBound.get().isBefore(rangeBound.get())) {
+            throw invalidRangeProperty(columnName, dateFormatPattern);
+        }
+
+        int interval = getProjectionPropertyValue(columnProperties, COLUMN_PROJECTION_INTERVAL, Integer.class::cast).orElse(1);
+        ChronoUnit intervalUnit = getProjectionPropertyValue(columnProperties, COLUMN_PROJECTION_INTERVAL_UNIT, ChronoUnit.class::cast)
+                .orElseGet(() -> resolveDefaultChronoUnit(columnName, dateFormatPattern));
+
+        if (!DATE_PROJECTION_INTERVAL_UNITS.contains(intervalUnit)) {
+            throw invalidProjectionException(
+                    columnName,
+                    format(
+                            "Property: '%s' value '%s' is invalid. Available options: %s",
+                            COLUMN_PROJECTION_INTERVAL_UNIT,
+                            intervalUnit,
+                            DATE_PROJECTION_INTERVAL_UNITS));
+        }
+
+        return new DateProjection(columnName, dateFormat, leftBound, rangeBound, interval, intervalUnit);
+    }
+
+    private ChronoUnit resolveDefaultChronoUnit(String columnName, String dateFormatPattern)
+    {
+        if (dateFormatPattern.contains("S") || dateFormatPattern.contains("s")
+                || dateFormatPattern.contains("m") || dateFormatPattern.contains("H")) {
+            throw invalidProjectionException(
+                    columnName,
+                    format(
+                            "Property: '%s' needs to be set when '%s' precision is more than day",
+                            COLUMN_PROJECTION_INTERVAL_UNIT,
+                            COLUMN_PROJECTION_FORMAT));
+        }
+        if (dateFormatPattern.contains("d")) {
+            return DAYS;
+        }
+        return MONTHS;
+    }
+
+    private Supplier<Instant> parseDateRangerBound(String columnName, String value, SimpleDateFormat dateFormat)
+    {
+        Matcher matcher = DATE_RANGE_BOUND_EXPRESSION_PATTERN.matcher(value);
+        if (matcher.find()) {
```

Comment:
`matcher.matches()`?  Probably needs allowing leading and trailing whitespace to pattern

Reply:
Applied. Whitespaces are already included in pattern